`chatTerminalToolProgressPart.ts` —  Trying to add a disposable to a DisposableStore

Seeing this after running some terminal commands in chat 

```
lifecycle.ts:477 Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!
    at r2i.add (lifecycle.ts:477:18)
    at _rt.D (lifecycle.ts:555:22)
    at _rt.P (chatTerminalToolProgressPart.ts:260:41)
    at s (chatTerminalToolProgressPart.ts:180:9)
    at _rt.L (chatTerminalToolProgressPart.ts:183:9)
    at async chatTerminalToolProgressPart.ts:124:4
```


Maybe the instance gets disposed of around these async calls: https://github.com/microsoft/vscode/blob/ebb5f678cb79d86b18de20cb8c4a2442a160a98f/src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts#L183

--- a/src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts
+++ b/src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts
@@ -174,7 +174,7 @@ export class ChatTerminalToolProgressPart extends BaseChatToolInvocationSubPart 
 		}
 
 		const attachInstance = async (instance: ITerminalInstance | undefined) => {
-			if (!instance || this._terminalInstance === instance) {
+			if (!instance || this._terminalInstance === instance || this._store.isDisposed) {
 				return;
 			}
 			this._terminalInstance = instance;

